<launch>
  <arg name="sensor_name" default="primesense" />
  <arg name="visualize" default="false" />
  <arg name="load_camera_info_from_ros_param" default="false" />

  <node name="depth_segmentation_node" pkg="depth_segmentation" type="depth_segmentation_node" output="screen">
    <rosparam command="load" file="$(find depth_segmentation)/cfg/$(arg sensor_name)_topics.yaml"/>
    <rosparam command="load" file="$(find depth_segmentation)/cfg/$(arg sensor_name)_config.yaml"/>
    <param name="semantic_instance_segmentation/enable" value="true"/>
    <param name="label_display" value="$(arg visualize)"/>
    <param name="load_camera_info_from_ros_param" value="$(arg load_camera_info_from_ros_param)"/>
    <rosparam command="load" file="$(find depth_segmentation)/cfg/$(arg sensor_name)_camera_info.yaml" if="$(arg load_camera_info_from_ros_param)"/>
  </node>
</launch>
